### ISSUE TITLE
[REVIEW]: PVGeo: an open-source Python package for geoscientific visualization in VTK and ParaView

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1451]
**Submitting author:** @banesullivan (<a href="http://orcid.org/0000-0001-8628-4566">C. Bane Sullivan</a>)
**Repository:** <a href="https://github.com/OpenGeoVis/PVGeo" target ="_blank">https://github.com/OpenGeoVis/PVGeo</a>
**Version:** v2.0.1
**Editor:** @leouieda
**Reviewer:** @gassmoeller, @leguark
**Archive:** <a href="https://doi.org/10.5281/zenodo.3261820" target="_blank">10.5281/zenodo.3261820</a>

## Status

[![status](http://joss.theoj.org/papers/1bd4f95c2944e187daf6c394240cde13/status.svg)](http://joss.theoj.org/papers/1bd4f95c2944e187daf6c394240cde13)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/1bd4f95c2944e187daf6c394240cde13"><img src="http://joss.theoj.org/papers/1bd4f95c2944e187daf6c394240cde13/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/1bd4f95c2944e187daf6c394240cde13/status.svg)](http://joss.theoj.org/papers/1bd4f95c2944e187daf6c394240cde13)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/OpenGeoVis/PVGeo" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@gassmoeller & @leguark, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @leouieda know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/OpenGeoVis/PVGeo">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v2.0.1
- [x] **Authorship:** Has the submitting author (@banesullivan) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/OpenGeoVis/PVGeo">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v2.0.1
- [x] **Authorship:** Has the submitting author (@banesullivan) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @gassmoeller, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01451/joss.01451/10.21105.joss.01451.pdf)

**Comment by banesullivan** [comment]
@whedon generate pdf from branch joss-patch-figures

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-patch-figures. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01451/joss.01451/10.21105.joss.01451.pdf)

**Comment by banesullivan** [comment]
@whedon generate pdf from branch joss-patch-figures-2

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-patch-figures-2. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01451/joss.01451/10.21105.joss.01451.pdf)

**Comment by leouieda** [comment]
:wave: Hi @gassmoeller @Leguark just checking in to see how the review is going. Please let me know if you have any questions about the process. And please remember to include `#openjournals/joss-reviews#1451` in any issues or PRs so that I can keep track of them here.

**Comment by Leguark** [comment]
Hi @leouieda and @banesullivan 

sorry for the delay. First of all, thank you for the authors for the great work on this library. After couple of years dealing myself with vtk what you have done here is going to facilitate enormously how visualize geodata in 3D.

- pip installation works without further issues
- All test passed with a satisfactory coverage
- Most of the methods at least a short explanation of the functionality. Ideally, the most exposed methods--i.e. those that a user must called--would benefit of a longer description of parameters, attributes and returns.
- All examples are working and very well documented which makes up for the lack of more complete docstrings.

In short, the library has already solid foundations to keep building functionality on top. I did not find any of the joos criteria missing. Therefore, I recommend the publication of this library and I am looking forward the future developments of this library and its integration with the rest of open geoscientific tools on the following months.

**Comment by gassmoeller** [comment]
Hi @leouieda and @banesullivan, I apologize for the slow process, I was at a developer meeting for one of my projects last week. I will finish the review until early next week.

**Comment by gassmoeller** [comment]
@banesullivan: I am not sure if it matters, but this issue still lists version 1.2.3 as the one for review, while you mentioned in #1384 that you would like to submit 2.0.0 for review. We should probably change the version in this issue.

**Comment by gassmoeller** [comment]
Dear @leouieda and @banesullivan

I have now finished my review of PVGeo and the accompanying publication in JOSS and can fully recommend its publication. I have verified that the software works as described and I am convinced that it addresses an important need for Geoscientists that can be well extended to more applications in the future. The repository structure is logical and follows best practices and the documentation allows users to understand its purpose and integrate it into their own workflow. The software follows best practices regarding testing, coding, and distribution. The JOSS manuscript clearly describes the need for and application of the software and references relevant projects and publications.

I have opened a minor issue that I would like to see addressed to make it simpler for new users to contribute to the development (see OpenGeoVis/PVGeo#49), and this review issue should be updated to reflect the authors request to submit version 2.0.0 instead of 1.2.3, but otherwise I recommend this paper for timely publication.

Sincerely,
Rene Gassmoeller

**Comment by banesullivan** [comment]
Hi @Leguark and @gassmoeller, thank you for the speedy reviews! I have been traveling lately and went offline for about a week so I'm happy to address any issues now.

> I have opened a minor issue that I would like to see addressed to make it simpler for new users to contribute to the development (see OpenGeoVis/PVGeo#49)

Thanks for the suggestion, @gassmoeller. I just updated PVGeo's readme to address this.

> this review issue should be updated to reflect the authors request to submit version 2.0.0 instead of 1.2.3

Ah yes, version 2.0.0 (actually 2.0.1 for when I tag and archive the repo for the submission)

**Comment by banesullivan** [comment]
Hi @leouieda - I'd like to make sure this hasn't fallen off the radar. Could we make a list of the remaining action items?

**Comment by leouieda** [comment]
@banesullivan sorry, it did fall off the radar for a while. Thanks for the reminder :slightly_smiling_face:

@gassmoeller and @Leguark thank you for the reviews! Regarding the version number, it's OK that there is a newer release. We'll update the version number here when it's time for publication. I see that you two **still have unchecked items** in your checklists. Is there anything that still needs to be addressed for those items? If not, please check them so we can move forward.

**Comment by Leguark** [comment]
No, I am simply not able to edit the first message for some reason but you can consider all checked

**Comment by gassmoeller** [comment]
I was just waiting for your confirmation about the version number. Everything ready from my side.

**Comment by leouieda** [comment]
@Leguark seems like Whedon un-assigned you temporarily from the issue for some reason. I've gone ahead and checked your boxes. 

@gassmoeller thanks for the confirmation!

**Comment by leouieda** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1016/S0098-3004(97)00032-0 is OK
- 10.21105/joss.01450 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by leouieda** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01451/joss.01451/10.21105.joss.01451.pdf)

**Comment by leouieda** [comment]
@banesullivan since both reviewers are satisfied with this submission I'm pleased to move ahead with acceptance :tada: 

Here are the steps you need to take now:

- Double check authors and affiliations (including ORCIDs)
- Do a final proof-reading of the paper and fix any outstanding issues (please link them to this issue)
- Make a release with the latest changes from the review and post the version number here.
- Archive the release on Zenodo/figshare and post the DOI here.

**Comment by leouieda** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01451/joss.01451/10.21105.joss.01451.pdf)

**Comment by banesullivan** [comment]
Latest version of the paper looks great!

**Comment by banesullivan** [comment]
@leouieda - releases are being pushed on version `v2.0.1`

Zenodo: [![DOI](https://zenodo.org/badge/97628892.svg)](https://zenodo.org/badge/latestdoi/97628892)

**Comment by leouieda** [comment]
@whedon set 10.5281/zenodo.3261820 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3261820" target="_blank">10.5281/zenodo.3261820</a> is the archive.

**Comment by leouieda** [comment]
@whedon set v2.0.1 as version

**Comment by whedon** [comment]
OK. v2.0.1 is the version.

**Comment by leouieda** [comment]
:wave: @openjournals/joss-eics I'm happy to say that this contribution is ready for publication!

@banesullivan congratulations on the acceptance :confetti_ball: 

@Leguark @gassmoeller thank you very much for you reviews! We appreciate your time and dedication. If you wish to be called upon for future reviews, please feel free to sign up to our reviewer pool at https://joss.theoj.org/reviewer-signup.html

**Comment by kyleniemeyer** [comment]
Hi @banesullivan, I'm about to look over the paper to do some last checks, but first, can you please clean up the Zenodo archive metadata? The list of authors should match that of the paper, and right now you have a few extra people listed.

**Comment by banesullivan** [comment]
Hi @kyleniemeyer, I just updated the Zenodo archives meta data

**Comment by kyleniemeyer** [comment]
@banesullivan I just submitted a small PR that fixes a few citation commands in the paper, could you merge that? https://github.com/OpenGeoVis/PVGeo/pull/52

Also, is there a better reference for VTK.js? At minimum, some names or a group should be included as the author. I'm also wondering if they have a preferred citation.

**Comment by kyleniemeyer** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01451/joss.01451/10.21105.joss.01451.pdf)

**Comment by banesullivan** [comment]
@kyleniemeyer 

> is there a better reference for VTK.js? At minimum, some names or a group should be included as the author. I'm also wondering if they have a preferred citation.

I'm not sure if there is another reference for VTK.js. Considering the VTK.js project's [README](https://github.com/Kitware/vtk-js/blob/master/README.md) states:
> vtk.js aims to be a subset of VTK

And points to the VTK Textbook reference that we currently have, maybe it would be fine to use the VTK reference?

**Comment by kyleniemeyer** [comment]
It's good to cite software directly, in addition to a published work. So perhaps in the text, when you mention VTK.js, you can cite the textbook in addition to the VTK.js software. For the software citation, perhaps add `author = {{Kitware}}`, and then add the version you use and the year it was released. If that is the current version, you can just use "version 8.11.1" and 2019.

**Comment by banesullivan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01451/joss.01451/10.21105.joss.01451.pdf)

**Comment by banesullivan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01451/joss.01451/10.21105.joss.01451.pdf)

**Comment by kyleniemeyer** [comment]
Ok, great!

**Comment by kyleniemeyer** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/804

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/804, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1016/S0098-3004(97)00032-0 is OK
- 10.21105/joss.01450 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by kyleniemeyer** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1145473468261117953) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/805
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01451](https://doi.org/10.21105/joss.01451)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by kyleniemeyer** [comment]
@banesullivan congratulations on your submission's publication in JOSS! Thanks to @gassmoeller and @Leguark for reviewing and @leouieda for editing.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.01451/status.svg)](https://doi.org/10.21105/joss.01451)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01451">
  <img src="http://joss.theoj.org/papers/10.21105/joss.01451/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.01451/status.svg
   :target: https://doi.org/10.21105/joss.01451
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.01451/status.svg)](https://doi.org/10.21105/joss.01451)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-joss
